### DISCUSSION TITLE
Is this format intended to be extensible, to support MIDI performance data for example?

### DISCUSSION THREAD

**Discussion by alindsay55661** [w3c/mnx#337] [Q&A] [answered]
Many have expressed a desire and need for formats that convey both musical information as traditionally viewed by humans through notation together with musical information (usually performance related) as viewed by machines through MIDI. For example, when expressing an eighth note with an accent, exact performance length and velocity can also be included as descriptors. Consumers that are able can then express notation and make use of performance data in the same file. Today, the merging of these two streams of musical data is awkwardly handled through the proprietary technology of various DAWs and notation programs. Many are asking and hoping for an open standard that facilitates coupling notation and performance.

So, my question is whether MNX is intended to allow for (i.e. invite) extensions that would make this possible, or if the format itself wants to limit the domain it covers to strictly static notation. Of course, being JSON, the schema can be extended easily in the wild, but this question is more one of intention and whether the working group is open to collaboration and/or potential adoption of extensions that describe music beyond the scope of traditional notation.

**Comment by adrianholovaty** [comment, marked as answer]
Thanks for the message/question. I'd expect that the initial stable version of MNX ("MNX 1.0") won't natively include that type of performance information, but we do indeed want to provide a way to extend the format with your own custom data.

Specifically regarding performance information: my personal bias would be to allow for the encoding of _multiple_ performances for the same piece of music, as that's something my own product supports. This would align nicely with the idea of multiple layouts, [already supported by MNX](https://w3c.github.io/mnx/docs/mnx-reference/examples/multiple-layouts/). But that'd likely be overkill for the vast majority of use cases, so hopefully (if/when we tackle this particular encoding problem) we'd find a balance between richness and usability.

**Reply by alindsay55661** [reply]
Fantastic, thanks for the answer @adrianholovaty, this is great news. It makes sense to support multiple layouts for things like condensing, grouping, parts vs. score, etc. I think this need is not as pronounced on the performance data side, but still quite relevant and good to make possible when implemented. For example, in an education context, a single file can contain multiple stylistic performances for teaching purposes (contrasting nuance, timing, "feel", interpretation of notation, etc.). It is quite useful to open the notation and then select from one or more performances for playback. Even something as basic as teaching how Jazz defaults differ from classical, or illustrating a range of swing strengths... all very useful. Glad to hear you are thinking about these use cases and encouraging a format that addresses modern realities.
